### DISCUSSION TITLE
How much VGA Ram (4 GB fails) is necessary for medium model? RuntimeError: CUDA out of memory. Tried to allocate

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [openai/whisper#60] [Q&A] [answered]
Hello everyone. I am in love with this release. Actually, this is the first time ever I have found an easy-to-install and actually working AI release to the public.

I have installed CUDA-enabled Pytorch on Windows 10 computer however when I try speech-to-text decoding with CUDA enabled it fails due to ram error

`RuntimeError: CUDA out of memory. Tried to allocate 70.00 MiB (GPU 0; 4.00 GiB total capacity; 2.87 GiB already allocated; 0 bytes free; 2.88 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF`

I disabled and enabled the graphic card before running the code - thus the VGA ram was 100% empty. Tested on my laptop so it has another GPU as well. Therefore, my GTX 1050 was literally using 0 MB of memory already

My computer also has 32 GB ram and CPU synthesis working very well but just too slow. In 7 hours processed only 1 hour of speech.

**Do you have any suggestions to solve this problem?**

![image](https://user-images.githubusercontent.com/19240467/191924611-09474700-038a-4938-a4e7-c49435edc73a.png)

**my pytorch version**

![image](https://user-images.githubusercontent.com/19240467/191925917-84afe087-57e2-4205-849a-9156a7225911.png)

**Comment by edgartaor** [comment, marked as answer]
Which model did you use? Try using a smaller model. With 4GB VRAM you should be able to use the small model.

whisper --model small audio.mp3

<html><body>


Size | Parameters | English-only model | Multilingual model | Required VRAM | Relative speed
-- | -- | -- | -- | -- | --
tiny | 39 M | tiny.en | tiny | ~1 GB | ~32x
base | 74 M | base.en | base | ~1 GB | ~16x
small | 244 M | small.en | small | ~2 GB | ~6x
medium | 769 M | medium.en | medium | ~5 GB | ~2x
large | 1550 M | N/A | large | ~10 GB | 1x


</body>
</html>

**Reply by FurkanGozukara** [reply]
Looks like i need more ram to use medium model. I ordered 12 gb VRAM having RTX 3060 today.

**Comment by mahi80** [comment]
#RUN in terminal windows 11 applicable for tiny model for 4GB GPU card and error torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 4.00 GiB total capacity; 3.21 GiB already allocated; 0 bytes free; 3.46 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.

set 'PYTORCH_CUDA_ALLOC_CONF=max_split_size_mb:38'  
devices = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
model = whisper.load_model("tiny", device = devices)

now it should run perfectly well
